libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -540,42 +538,20 @@ object MapboxRouteLineUtils {
      * @return either the layer ID if found else a default layer ID
      */
     @JvmStatic
-    fun getDefaultBelowLayer(layerId: String?, style: Style): String {
-        val layers = style.styleLayers
-        return when (layerId.isNullOrEmpty()) {
-            false -> checkLayerIdPresent(layerId, layers)
-            true -> findLayerBelow(layers)
-        }
-    }
-
-    private fun checkLayerIdPresent(layerId: String, layers: List<StyleObjectInfo>): String {
-        val foundId = layers.firstOrNull { it.id == layerId }?.id
-        if (foundId == null) {
-            Timber.e(
-                """Tried placing route line below "$layerId" which doesn't exist"""
-            )
-        }
-        return foundId ?: LocationComponentConstants.MODEL_LAYER
-    }
-
-    /**
-     * Tries to find a reference layer ID that's above a first non-symbol layer from the top
-     * of the stack of layers. Additionally, the algorithm always ensures that the reference
-     * layer is below the puck layers.
-     */
-    private fun findLayerBelow(layers: List<StyleObjectInfo>): String {
-        val puckLayerIndex = layers.indexOfFirst {
-            it.id.contains(RouteConstants.MAPBOX_LOCATION_ID)
-        }
-        val lastSymbolLayerFromTopIndex = layers.indexOfLast {
-            it.type != "symbol" && !it.id.contains(RouteConstants.MAPBOX_LOCATION_ID)
-        } + 1
-        val index = if (puckLayerIndex in 0 until lastSymbolLayerFromTopIndex) {
-            puckLayerIndex
-        } else {
-            lastSymbolLayerFromTopIndex
+    fun getBelowLayerIdToUse(belowLayerId: String?, style: Style): String? {
+        return when (style.styleLayers.firstOrNull { it.id == belowLayerId } != null) {
+            true -> belowLayerId
+            false -> {
+                when (
+                    style.styleLayers.firstOrNull {
+                        it.id == LocationComponentConstants.MODEL_LAYER
+                    } != null
+                ) {
+                    true -> LocationComponentConstants.MODEL_LAYER
+                    false -> null
+                }
+            }
```

Comment:
The more I think about it, the more of a hack and undefined behavior it seems to me from the user's perspective. This depends on an order of initialization and produces different results depending on whether the puck was initialized first or the route. This would be very tricky for the developer unfamiliar with our stack to debug and understand why sometimes the route is below the puck and sometimes it isn't.

I'm starting to lean towards:
```suggestion
    fun getBelowLayerIdToUse(belowLayerId: String?, style: Style): String? {
        return when (style.styleLayersExists(belowLayerId)) {
            true -> belowLayerId
            false -> null
```

and instead clearly documenting in both class level docs and options builder docs that `LocationComponentConstants.MODEL_LAYER` is the preferred value for the optimized experience but **it's not** a default value. Developers would have to discover it themselves, but that should be quick after the route is placed on top of everything when they first build the project.

Sorry for the back-and-forth on this, but it's been causing a lot of frustration in the past and I'd like to find a clearer solution. What do you think @cafesilencio?

/cc @1ec5 for thoughts on how this is implemented on iOS and whether it makes sense to align

Reply:
cc @jill-cardamon since I heard you're working on route line z-indexing bugs on iOS

Reply:
I agree. I think there are too many variables to try and come up with an opinionated solution.  It's better to point people in the right direction.